node/aggregate/aggregate.go

```diff
@@ -8,45 +8,39 @@ import (
 	"github.com/blocklessnetwork/b7s/models/execute"
 )
 
-type Results []Result
-
-// Result represents the execution result along with its aggregation stats.
-type Result struct {
-	Result execute.RuntimeOutput `json:"result,omitempty"`
-	// Peers that got this result.
-	Peers []peer.ID `json:"peers,omitempty"`
-	// How frequent was this result, in percentages.
-	Frequency float64 `json:"frequency,omitempty"`
-}
-
-type resultStats struct {
-	seen  uint
-	peers []peer.ID
-}
-
 func Aggregate(results execute.ResultMap) Results {
 
 	total := len(results)
 	if total == 0 {
 		return nil
 	}
 
+	type resultStats struct {
+		seen     uint
+		peers    []peer.ID
+		metadata map[peer.ID]any
```

Comment:
not all peers see the same metadata; hence i dont think we should make this a map.
Instead `any` should be sufficient?
This also makes metadata extraction from the result a bit difficult; since we'd need access to the peer-id.

Reply:
Perhaps I misunderstood something. From the way things were set before the following was also true - metadata can be set on node arbitrarily. So worker A produces metadataA, worker B produces metadataB and so on.

When we run an execution request, we get multiple execution results back. They might be the same, they might not. Same goes for metadata.

How would we determine which metadata to return when we have five workers? In the other PR it was random - [the last seen execution result overrides any previous metadata](https://github.com/blocklessnetwork/b7s/blob/feat/worker-response-metadata-support/node/aggregate/aggregate.go#L54). I don't feel that's correct, unless the use case is some that I don't understand.

We mentioned signatures before - if signature is part of the metadata, it's typically tied to the workers identity, so it will definitely be different for each worker. If we have results of workers 1 through 7 and have a single metadata object that has `signature3` - what do we do with it?